include/gatekeeper_ggu.h

```diff
@@ -19,11 +19,66 @@
 #ifndef _GATEKEEPER_GGU_H_
 #define _GATEKEEPER_GGU_H_
 
+#include "gatekeeper_net.h"
+#include "gatekeeper_flow.h"
+
 /* Configuration for the GK-GT Unit functional block. */
 struct ggu_config {
-	unsigned int	lcore_id;
+	unsigned int      lcore_id;
+
+	/* The UDP source and destination port numbers for GGU. */
+	uint16_t          ggu_src_port;
+	uint16_t          ggu_dst_port;
+
+	/*
+	 * The fields below are for internal use.
+	 * Configuration files should not refer to them.
+	 */
+
+	/* RX queue on the back interface. */
+	uint16_t          rx_queue_back;
+	struct net_config *net;
+	struct gk_config  *gk;
+};
+
+/*
+ * Since the length of IPv6 policy is much larger than IPv4 policy,
+ * to save network bandwidth, we choose to use different data structures
+ * to store the in-packet policies for IPv4 and IPv6, respectively.
+ *
+ * Packet format: Ethernet headers + IP header + UDP header + Data.
+ * In the IP payload, the format is: num_policies (8 bits) + proto1 + proto2
+ * + ... + policy1 + policy2 + ... .
```

Comment:
I think this version will definitely make the decisions compact. A minor disadvantage is that the GT needs more work to fill up the packets with these decisions, but it's reasonable.